hypershift-operator/controllers/nodepool/haproxy.go

```diff
@@ -66,7 +66,7 @@ func (r *NodePoolReconciler) reconcileHAProxyIgnitionConfig(ctx context.Context,
 
 	if util.IsPrivateHC(hcluster) {
 		apiServerExternalAddress = fmt.Sprintf("api.%s.hypershift.local", hcluster.Name)
-		apiServerExternalPort = 443
+		apiServerExternalPort = util.KASPodPortForPrivateHostedCluster(util.IsRouteKASByHC(hcluster))
```

Comment:
why would external port come from KAS pod?

Reply:
It appears to me that this port (used when the `api.<cluster-name>.hypershift.local` endpoint) is the same as the one of the Route/LB which exposes the KAS on the MC side. This would be 443 for route KAS or 6443 for LB KAS. Hence this line.

Reply:
i see, that's different from the pod port.